src/network/core/tcp_connect.cpp

```diff
@@ -371,18 +371,14 @@ bool TCPConnecter::CheckActivity()
 			closesocket(*it);
 			this->sock_to_address.erase(*it);
 			it = this->sockets.erase(it);
+			n--;
```

Comment:
I don't understand this.
`n` is the number of bits set in `write_fd`. This loop does not check `write_fd`.
What says that the sockets with error are new errors?

Reply:
There are only 2 possible ways that write_fd gets set: either the socket is connected or is in error. That is why this work. But also why I wrote the lines in "limitations" :)

Reply:
Ah, I missed that the loop removed failed sockets from `sockets`. I was wondering whether there were still sockets which errored on the previous call.